src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1,1161 +1,608 @@
-
-```astro mark="Astro.generator"
-<html>
-  <head>
-    <meta name="generator" content={Astro.generator} />
-  </head>
-  <body>
-    <footer>
-      <p>Built with <a href="https://astro.build">{Astro.generator}</a></p>
-    </footer>
-  </body>
-</html>
-```
-
-### `Astro.slots`
-
-`Astro.slots` contains utility functions for modifying an Astro component's slotted children.
-
-#### `Astro.slots.has()`
-
-<p>
-
-**Type:** `(slotName: string) => boolean`
-</p>
-
-You can check whether content for a specific slot name exists with `Astro.slots.has()`. This can be useful when you want to wrap slot contents, but only want to render the wrapper elements when the slot is being used.
+The `Astro` global object is available to all `.astro` files. Use the `context` object in [endpoint functions](/en/guides/endpoints/) to serve static or live server endpoints and in [middleware](/en/guides/middleware/) to inject behavior when a page or end point is about to be rendered.
 
-```astro  title="src/pages/index.astro"
----
----
-<slot />
-
-{Astro.slots.has('more') && (
-  <aside>
-    <h2>More</h2>
-    <slot name="more" />
-  </aside>
-)}
-```
-
-#### `Astro.slots.render()`
-
-<p>
-
-**Type:** `(slotName: string, args?: any[]) => Promise<string>`
-</p>
+## The context object
 
-You can asynchronously render the contents of a slot to a string of HTML using `Astro.slots.render()`.
-
-```astro
----
-const html = await Astro.slots.render('default');
----
-<Fragment set:html={html} />
-```
-
-:::note
-This is for advanced use cases! In most circumstances, it is simpler to render slot contents with [the `<slot />` element](/en/basics/astro-components/#slots).
-:::
+The following items are available on the `Astro` global (e.g. `Astro.props`, `Astro.redirect()`) and most are also available on an endpoint context object (e.g. `context.props`, `context.redirect()`).
 
-`Astro.slots.render()` optionally accepts a second argument: an array of parameters that will be forwarded to any function children. This can be useful for custom utility components.
+### `props`
 
-For example, this `<Shout />` component converts its `message` prop to uppercase and passes it to the default slot:
+`props` is an object containing any values that have been passed as [component attributes](/en/basics/astro-components/#component-props).
 
-```astro title="src/components/Shout.astro" "await Astro.slots.render('default', [message])"
+```astro {3}
 ---
-const message = Astro.props.message.toUpperCase();
-let html = '';
-if (Astro.slots.has('default')) {
-  html = await Astro.slots.render('default', [message]);
-}
+// src/components/Heading.astro
+const { title, date } = Astro.props;
 ---
-<Fragment set:html={html} />
+<div>
+  <h1>{title}</h1>
+  <p>{date}</p>
+</div>
 ```
 
-A callback function passed as `<Shout />`’s child will receive the all-caps `message` parameter:
-
-```astro title="src/pages/index.astro"
+{/* ```astro /title=".+"/ /date=".+"/ */}
+```astro "title=" "date="
```

Comment:
We can do something like this if we want the contents highlighted too.
``````suggestion
```astro /(?:title|date)="(?:\\"|.)*?"/
``````

Reply:
I actually changed this because I *didn't* want the contents highlighted. 😅  Do you think it looks better with?

Reply:
nope, looks fine as is! was just confused by the comment. lgtm!